apps/spruce/src/components/Table/BaseTable.tsx

```diff
@@ -80,6 +86,18 @@ export const BaseTable = forwardRef(
     // @ts-expect-error: FIXME. This comment was added by an automated script.
     const { rows } = table.getRowModel();
     const hasVirtualRows = virtualRows && virtualRows.length > 0;
+
+    // Auto-redirect user to first page when filter applied.
+    useEffect(() => {
+      if (
+        table &&
+        table.getRowModel().rows.length === 0 &&
+        table.getRowCount() !== 0
+      ) {
+        table.firstPage();
+      }
+    }, [table]);
```

Comment:
rather than using `useEffect`, can you return to the first page exactly when the filter is applied?